src/frequenz/dispatch/actor.py

```diff
@@ -114,18 +89,28 @@ async def _fetch(self) -> None:
 
         try:
             _logger.info("Fetching dispatches for microgrid %s", self._microgrid_id)
-            async for dispatch in self._client.list(microgrid_id=self._microgrid_id):
-                self._dispatches[dispatch.id] = dispatch
+            async for client_dispatch in self._client.list(
+                microgrid_id=self._microgrid_id
+            ):
+                dispatch = Dispatch(client_dispatch)
 
+                self._dispatches[dispatch.id] = Dispatch(client_dispatch)
                 old_dispatch = old_dispatches.pop(dispatch.id, None)
```

Comment:
Be careful with this, if the actor crashes, you might lose this dispatch in the list of `old_dispatches`, not sure if it would be a problem or not.

Reply:
Hmm it might not be a problem to just re-run all dispatches that are currently supposed to be running?